kong/api/routes/consumers.lua

```diff
@@ -10,11 +10,18 @@ return {
   ["/consumers"] = {
     GET = function(self, db, helpers, parent)
       local args = self.args.uri
+      local custom_id = args.custom_id
+
+      if custom_id == "" then
+        return kong.response.exit(400, {
+          message = "custom_id cannot be empty string",
+        })
+      end
 
       -- Search by custom_id: /consumers?custom_id=xxx
-      if args.custom_id then
+      if custom_id then
```

Comment:
Alternatively here we could just do:

```
if custom_id and custom_id ~= "" then
...
```


Aka empty custom_id == no custom id at all?

Reply:
Also I don't fully understand why it crashes with empty string. Could it then just return 404 instead?

Reply:
@bungle My original idea is to ignore the empty string. However, that is not proper: https://github.com/Kong/kong/pull/10475#discussion_r1136117260.
The empty string would fail the argument check, and the check throws a Lua error instead of returning a err string.